### PR TITLE
Add Speaker embedding models to contrib

### PR DESCRIPTION
**Pull request by TCord**
Adds basic ResNet-based speaker embedding extractor, helper functions for training of speaker embedding extractors, and framework for teacher-student training

### CODE REVIEW

File: padertorch/contrib/tcl/speaker_embeddings/loss.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+        for _, module in self.fc.named_modules():
+            if isinstance(module, torch.nn.Linear):
+                module.weight.data = torch.nn.functional.normalize(module.weight, p=2, dim=1)
+
+        logits = torch.nn.functional.normalize(embeddings, p=2, dim=1)
+        logits = self.fc(logits)
+
+        if self.loss_type == 'aam':
+            numerator = self.s * (torch.diagonal(logits.transpose(0, 1)[labels]) - self.m)
```
Comment (boeddeker) [inline padertorch/contrib/tcl/speaker_embeddings/loss.py:75]: Could you swap the transpose arguments? It's then easier to read, because it matches numpy.

File: padertorch/contrib/tcl/utils/augmentation.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+from functools import partial
+import padertorch as pt
+
+
+class AugmentationHelper:
+    def __init__(self,
+                 augmentation_sets: Dict = None,
+                 p_augment: float = 0.,
+                 p_reverb=None,
+                 augmentation_type: Union[str, Iterable] = ('noise', 'music', 'speech'),
```
Comment (boeddeker) [inline padertorch/contrib/tcl/utils/augmentation.py:16]: Add a hint, for the values? e.g. 'reverb seems to be allowed'

File: padertorch/contrib/tcl/speaker_embeddings/loss.py (hunk 1 above)
Comment (TCord) [reply padertorch/contrib/tcl/speaker_embeddings/loss.py]: Done

File: padertorch/contrib/tcl/utils/augmentation.py (hunk 2 above)
Comment (TCord) [reply padertorch/contrib/tcl/utils/augmentation.py]: Added a docstring
